src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer.Common/ProjectEngineFactory.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.CodeAnalysis.Razor;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Common
+{
+    internal abstract class ProjectEngineFactory : IProjectEngineFactory
```

Comment:
📝 I decided that the various `ProjectEngineFactory_X_X` classes probably shouldn't just be copy-pastes of one another. Since I had to fix the fact that `Assembly.FullName` returns a `string?` on .NET 7, I went ahead and shared the implementation.